qlib/data/ops.py

```diff
@@ -1154,18 +1148,32 @@ class Rank(Rolling):
 
     def __init__(self, feature, N):
         super(Rank, self).__init__(feature, N, "rank")
+        major_version, minor_version, *_ = pd.__version__.split(".")
+        self._load_internal = (
```

Comment:
I would recommend implementing it as a method to override the parent method instead of setting the attribute for the following reasons.
- fewer attributes will make it simpler.
- It will not work in some special cases (for example, someone dump an instance of the operator in low version of pandas and load it in high version of pandas)

Reply:
@you-n-g
 
think of a way to use `hasattr` instead of version check. See latest commit. let me know if you have better idea. thx.

Reply:
Any update/feedback please?